Kernel/FileSystem/FATFS/Inode.cpp

```diff
@@ -164,14 +164,14 @@ ErrorOr<NonnullOwnPtr<KBuffer>> FATInode::read_block_list()
 
     auto builder = TRY(KBufferBuilder::try_create());
 
-    u8 buffer[512];
-    VERIFY(fs().m_device_block_size <= sizeof(buffer));
-    auto buf = UserOrKernelBuffer::for_kernel_buffer(buffer);
+    Array<u8, 512> buffer {};
+    VERIFY(fs().m_device_block_size <= buffer.span().size());
```

Comment:
Just `buffer.size()` should work?

Reply:
That's me being a bit paranoid here because `buffer.size()` could work, but if you have an `Array` of something not `u8`, then using this is not correct anymore, while `buffer.span().size())` will always give the amount of bytes, not the amount of entries.

Reply:
`buffer.span().size()` does _not_ yield the size in bytes, it is exactly equivalent to `buffer.size()`.

If one needs the size in bytes then one should use `sizeof`.